fe/fe-core/src/main/java/com/starrocks/catalog/CatalogRecycleBin.java

```diff
@@ -263,14 +253,7 @@ public synchronized short getPartitionReplicationNum(long partitionId) {
 
     public synchronized Range<PartitionKey> getPartitionRange(long partitionId) {
         RecyclePartitionInfo partitionInfo = idToPartition.get(partitionId);
-        if (partitionInfo != null) {
-            if (partitionInfo instanceof RecyclePartitionInfoV1) {
-                return ((RecyclePartitionInfoV1) partitionInfo).getRange();
-            } else {
-                return ((RecycleRangePartitionInfo) partitionInfo).getRange();
-            }
-        }
-        return null;
+        return partitionInfo != null ? partitionInfo.getRange() : null;
```

Comment:
All the getPartition related function needs to filter the unrecoverable partition.